### PR TITLE
Introduce util methods for HTTP PUT requests

### PR DESCRIPTION
**Pull request by ahennr**
This PR introduces methods for HTTP PUT requests to `HttpUtil` and their corresponding tests.

### CODE REVIEW

File: src/shogun2-util/src/main/java/de/terrestris/shogun2/util/http/HttpUtil.java [hunk 1]
```diff
@@ -486,6 +488,138 @@ private static Response postParams(HttpPost httpRequest,
...
 	/**
+	 * Perform HTTP PUT with empty body
+	 *
+	 * @param uri
+	 * @return
+	 * @throws HttpException
+	 * @throws URISyntaxException
+	 */
+	public static Response put(URI uri) throws URISyntaxException, HttpException {
+		return putBody(new HttpPut(uri), StringUtils.EMPTY, ContentType.APPLICATION_JSON, null, null);
```
Comment (dnlkoch) [inline src/shogun2-util/src/main/java/de/terrestris/shogun2/util/http/HttpUtil.java]: Why do we set `ContentType.APPLICATION_JSON` here (and in the ones below) without having a body content?

File: src/shogun2-util/src/main/java/de/terrestris/shogun2/util/http/HttpUtil.java [hunk 2]
```diff
@@ -486,6 +488,138 @@ private static Response postParams(HttpPost httpRequest,
...
+	 * @param username
+	 * @param password
+	 * @return The HTTP response as Response object.
+	 * @throws URISyntaxException
+	 * @throws HttpException
+	 */
+	private static Response putBody(HttpPut httpRequest, String body, ContentType contentType, String username, String password) throws URISyntaxException, HttpException {
+		StringEntity stringEntity = new StringEntity(body, contentType);
+		stringEntity.setChunked(true);
+		httpRequest.setEntity(stringEntity);
```
Comment (dnlkoch) [inline src/shogun2-util/src/main/java/de/terrestris/shogun2/util/http/HttpUtil.java]: Maybe you can introduce a condition here that checks for a not-null ContentType.

**Comment by dnlkoch** [comment]
Very cool and useful addition! Thanks!

File: src/shogun2-util/src/main/java/de/terrestris/shogun2/util/http/HttpUtil.java (hunk 1 above)
Comment (ahennr) [inline src/shogun2-util/src/main/java/de/terrestris/shogun2/util/http/HttpUtil.java]: According to issue in method `putBody(...)` further down, will pass `null` here for `body` and `ContentType`.

File: src/shogun2-util/src/main/java/de/terrestris/shogun2/util/http/HttpUtil.java (hunk 2 above)
Comment (ahennr) [inline src/shogun2-util/src/main/java/de/terrestris/shogun2/util/http/HttpUtil.java]: Here, a body only will be included in the `HttpPut` request if it isn't `null` and has a valid `ContentType`

**Comment by dnlkoch** [comment]
Great! Feel free to merge!

**Comment by ahennr** [comment]
Thx for review, I'll merge!

**Pull request merged**